EasyLM/checkpoint.py

```diff
@@ -66,8 +66,9 @@ def save_pickle(self, obj, filename):
             path = '/dev/null'
         mlxu.save_pickle(obj, path)
 
-    def save_all(self, train_state, gather_fns, metadata=None, dataset=None, milestone=False, is_value=False):
-        step = int(jax.device_get(train_state.step))
+    def save_all(self, train_state, gather_fns, metadata=None, dataset=None, milestone=False, step=None, is_value=False):
+        if step is None:
+            step = int(jax.device_get(train_state.step))
```

Comment:
Otherwise the checkpointer would count steps as the number of backward passes, which is quite confusing ...